src/server/actions/user-keys.actions.ts

```diff
@@ -35,4 +36,22 @@ export const setReviewerPublicKeyAction = userAction(async ({ publicKey, fingerp
             fingerprint,
         })
         .executeTakeFirstOrThrow(() => new ActionFailure({ message: 'Failed to set reviewer public key' }))
+
+    onUserPublicKeyCreated({ userId })
+}, setOrgUserPublicKeySchema)
+
+export const updateReviewerPublicKeyAction = userAction(async ({ publicKey, fingerprint }) => {
+    const userId = await getUserIdFromActionContext()
+
+    await db
+        .updateTable('userPublicKey')
+        .set({
+            publicKey: Buffer.from(publicKey),
+            fingerprint,
+            updatedAt: new Date(),
+        })
+        .where('userId', '=', userId)
+        .executeTakeFirstOrThrow(() => new ActionFailure({ message: 'Failed to update reviewer public key.' }))
```

Comment:
the property in the ActionFailure object will look for a form input with a matching name to display errors on, or if not will capitalize it and make a string shown in a toast.  This will end up being something like: `Message Failed to update reviewer public key.` which reads a little awkward.  Maybe it'd be better to use `error: key did not update in database` so it reads `Error key did not update..`
```suggestion
        .executeTakeFirstOrThrow(() => new ActionFailure({ error: 'public key was not saved.' }))
```